p2p/host/peerstore/pstoremem/addr_book.go

```diff
@@ -109,7 +163,7 @@ func WithClock(clock clock) AddrBookOption {
 // background periodically schedules a gc
 func (mab *memoryAddrBook) background(ctx context.Context) {
 	defer mab.refCount.Done()
-	ticker := time.NewTicker(1 * time.Hour)
+	ticker := time.NewTicker(1 * time.Minute)
 	defer ticker.Stop()
```

Comment:
Can you replace this with a timer and a signal channel? That would mean we don't have to run this every minute and simply run it when the first element of the heap expires.

Reply:
I like the idea. I've implemented it on the branch `marco/better-peerstore-gc-and-timer`. Testing it though requires a bit more changes to use our mockclock instead of the stdlib timer and then updating the existing tests to use it. Doable, but a bit of a hassle.

Let's do it in a follow up PR.